// @iota/dapp-kit@0.5.0/dist/cjs/components/AccountDropdownMenu.css.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/components/AccountDropdownMenu.css.ts
var AccountDropdownMenu_css_exports = {};
__export(AccountDropdownMenu_css_exports, {
  connectedAccount: () => connectedAccount,
  menuContainer: () => menuContainer,
  menuContent: () => menuContent,
  menuItem: () => menuItem,
  scrollableContent: () => scrollableContent,
  separator: () => separator,
  switchAccountMenuItem: () => switchAccountMenuItem
});
module.exports = __toCommonJS(AccountDropdownMenu_css_exports);
var connectedAccount = "AccountDropdownMenu_connectedAccount__1x0gmpf0";
var menuContainer = "AccountDropdownMenu_menuContainer__1x0gmpf1";
var menuContent = "AccountDropdownMenu_menuContent__1x0gmpf2";
var menuItem = "AccountDropdownMenu_menuItem__1x0gmpf4";
var scrollableContent = "AccountDropdownMenu_scrollableContent__1x0gmpf3";
var separator = "AccountDropdownMenu_separator__1x0gmpf6";
var switchAccountMenuItem = "AccountDropdownMenu_switchAccountMenuItem__1x0gmpf5";
