// @iota/dapp-kit@0.5.0/dist/cjs/hooks/useIotaClientQueries.js
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/hooks/useIotaClientQueries.ts
var useIotaClientQueries_exports = {};
__export(useIotaClientQueries_exports, {
  useIotaClientQueries: () => useIotaClientQueries
});
module.exports = __toCommonJS(useIotaClientQueries_exports);
var import_react_query = require("@tanstack/react-query");

// src/hooks/useIotaClient.ts
var import_react2 = require("react");

// src/components/IotaClientProvider.tsx
var import_client = require("@iota/iota-sdk/client");
var import_react = require("react");
var import_jsx_runtime = require("react/jsx-runtime");
var IotaClientContext = (0, import_react.createContext)(null);
var DEFAULT_NETWORKS = {
  localnet: { url: (0, import_client.getNetwork)(import_client.Network.Localnet)?.url }
};

// src/hooks/useIotaClient.ts
function useIotaClientContext() {
  const iotaClient = (0, import_react2.useContext)(IotaClientContext);
  if (!iotaClient) {
    throw new Error(
      "Could not find IotaClientContext. Ensure that you have set up the IotaClientProvider"
    );
  }
  return iotaClient;
}

// src/hooks/useIotaClientQueries.ts
function useIotaClientQueries({
  queries,
  combine
}) {
  const iotaContext = useIotaClientContext();
  return (0, import_react_query.useQueries)({
    combine,
    queries: queries.map((query) => {
      const { method, params, options: { queryKey = [], ...restOptions } = {} } = query;
      return {
        ...restOptions,
        queryKey: [iotaContext.network, method, params, ...queryKey],
        queryFn: async () => {
          return await iotaContext.client[method](params);
        }
      };
    })
  });
}
